lib/src/simple_token.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library angular_ast.src.simple_token;
+
+import 'package:quiver/core.dart';
+
+part 'token/simple_type.dart';
+
+class NgSimpleToken {
+  factory NgSimpleToken.bang(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.bang, offset);
+  }
+
+  //Probably don't need
+  factory NgSimpleToken.closeBrace(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeBrace, offset);
+  }
+
+  factory NgSimpleToken.closeBracket(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeBracket, offset);
+  }
+
+  factory NgSimpleToken.closeParen(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeParen, offset);
+  }
+
+  factory NgSimpleToken.commentBegin(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.commentBegin, offset);
+  }
+
+  factory NgSimpleToken.commentEnd(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.commentEnd, offset);
+  }
+
+  factory NgSimpleToken.dash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.dash, offset);
+  }
+
+  factory NgSimpleToken.doubleQuote(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.doubleQuote, offset);
+  }
+
+  factory NgSimpleToken.doubleQuotedText(int offset, String lexeme) {
+    return new NgSimpleToken(
+        new NgSimpleTokenType.doubleQuotedText(lexeme), offset);
+  }
+
+  factory NgSimpleToken.tagStart(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.tagStart, offset);
+  }
+
+  factory NgSimpleToken.tagEnd(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.tagEnd, offset);
+  }
+
+  factory NgSimpleToken.EOF(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.EOF, offset);
+  }
+
+  factory NgSimpleToken.equalSign(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.equalSign, offset);
+  }
+
+  factory NgSimpleToken.forwardSlash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.forwardSlash, offset);
+  }
+
+  factory NgSimpleToken.hash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.hash, offset);
+  }
+
+  //Probably don't need
+  factory NgSimpleToken.openBrace(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openBrace, offset);
+  }
+
+  factory NgSimpleToken.openBracket(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openBracket, offset);
+  }
+
+  factory NgSimpleToken.openParen(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openParen, offset);
+  }
+
+  factory NgSimpleToken.singleQuote(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.singleQuote, offset);
+  }
+
+  factory NgSimpleToken.singleQuotedText(int offset, String lexeme) {
```

Comment:
Quotes could probably be an extension of `NgSimpleToken`. It can contain `openQuoteOffset` and `closedQuoteOffset` and even `isDoubleQuote`. That way we don't need to track six types of tokens (quote, closed quote, and unclosed quote all in single & double varieties) but we still don't throw out any information.